fairseq/checkpoint_utils.py

```diff
@@ -105,7 +105,7 @@ def get_checkpoint_path(args):
 def load_checkpoint(args, trainer):
     """Load a checkpoint and restore the training iterator."""
     # only one worker should attempt to create the required dir
-    if args.distributed_rank == 0:
+    if args.distributed_rank == 0 or xm.is_master_ordinal():
```

Comment:
Why keeping the original check?

Reply:
> Didn't want to override args.distributed_rank == 0 since that may break some other code path.

I thought it may break some other code path, like inference or gpu compatibility (<rant> which I really want to stop worrying about </rant>)